gpt_index/langchain_helpers/text_splitter.py

```diff
@@ -103,15 +103,32 @@ def _process_splits(self, splits: List[str]) -> List[str]:
                 new_splits.extend(cur_splits2)
         return new_splits
 
-    def split_text(self, text: str) -> List[str]:
+    def split_text(self, text: str, extra_info_str: Optional[str] = None) -> List[str]:
         """Split incoming text and return chunks."""
-        text_slits = self.split_text_with_overlaps(text)
+        text_slits = self.split_text_with_overlaps(text, extra_info_str=extra_info_str)
         return [text_split.text_chunk for text_split in text_slits]
 
-    def split_text_with_overlaps(self, text: str) -> List[TextSplit]:
+    def split_text_with_overlaps(
+        self, text: str, extra_info_str: Optional[str] = None
+    ) -> List[TextSplit]:
         """Split incoming text and return chunks with overlap size."""
         if text == "":
             return []
+
+        # NOTE: Consider extra info str that will be added to the chunk at query time
+        #       This reduces the effective chunk size that we can have
+        if extra_info_str is not None:
+            # NOTE: extra 2 newline chars for formatting when prepending in query
+            num_extra_tokens = len(self.tokenizer(f"{extra_info_str}\n\n")) + 1
```

Comment:
out of curiosity: why + 1?

Reply:
seems like e.g.  `apple\n\n` is 3 tokens, but `apple\n\napple` is 5 tokens. I'm not familiar enough with tokenization to know why 😂

Reply:
Hm ok. can you leave a TODO? the tokenizer is a bit of a black box. i imagine if there's a token exceeded bug later we may have to dig into this a bit more, but maybe we don't have to worry for now

Reply:
going to merge this for now to unblock